### ISSUE TITLE
Documentation: Design pattern for dynamic table loading spinner

### ISSUE THREAD

**Issue by ErnieAtLYD** [rackerlabs/encore-ui#1836]
@arc6789 developed some functionality required by PMs for the billing portion of Encore, in this case a spinner for the transactions tab: https://github.com/rackerlabs/billing-ui/pull/984

As this is a new design pattern through all of Encore, my goal is to submit a PR to the documentation which illustrates this pattern.

**Comment by parlarjb** [comment]
@ErnieAtLYD The framework does have this pattern already. The `rx-loading-overlay` can be used in your situation (I think).

You can see an example of it at http://rackerlabs.github.io/encore-ui/#/elements/Tables#api-based-pagination 

Specifically, click on page 6 in the pagination area, and you should see a loading overlay cover the table.

**Comment by arc6789** [comment]
We wanted a loading spinner just for table data (but still displaying table headers). In many cases we are on a page staring at a blank table while waiting for data to load for a very long time. It would be better to give the user some context on what data is about to load (hence decision to keep table headers).
I believe rx-loading-overlay is pagination based and also hides the entire page with the overlaying spinner?

Here is Nando’s PR, which is a follow up to my PR. https://github.com/rackerlabs/billing-ui/pull/989

**Comment by ErnieAtLYD** [comment]
This is the code for the rxLoadingOverlay:
https://github.com/rackerlabs/encore-ui/blob/master/src/elements/Tables/scripts/rxLoadingOverlay.js

Maybe it might make sense to slightly modify the rxLoadingOverlay directive to include the headers? Or include options?

**Comment by TheNando** [comment]
We looked at this, but there is no easy way to transclude the table headers while also hiding the footer and body until the promise resolves without significantly rewriting the existing directive. Also, in order to get the well to render at the appropriate size, we are including a column span attribute. We could not figure out another way to get it to nicely render how we intended. And finally, our directive is a little more general purpose as it works with any boolean equivalent expression instead of just promises.
